packages/wdio-config/src/lib/ConfigParser.ts

```diff
@@ -316,37 +327,58 @@ export default class ConfigParser {
     /**
      * returns a flatten list of globed files
      *
-     * @param  {String[]} filenames  list of files to glob
-     * @return {String[]} list of files
+     * @param  {String[] | String[][]} filenames list of files to glob
+     * @param  {Boolean} flag to indicate omission of warnings
+     * @param  {FileSystemPathService} file system path service for expanding globbed file names
+     * @param  {number} hierarchy depth to prevent recursive calling beyond a depth of 1
+     * @return {String[] | String[][]} list of files
      */
-    static getFilePaths (patterns: string[], omitWarnings?: boolean, findAndGlob: CurrentPathFinder & Globber & DeterminesAbsolutePath = new FileSystemPathService()) {
-        let files: string[] = []
+    static getFilePaths (patterns: (string | string[])[], omitWarnings?: boolean, findAndGlob: CurrentPathFinder & Globber & DeterminesAbsolutePath = new FileSystemPathService(), hierarchyDepth?: number) {
+        let files: (string | string[])[] = []
+        let groupedFiles: string[] = []
+        let depth: number = hierarchyDepth || 0
 
+        // Convert a string into an array as a convenience
         if (typeof patterns === 'string') {
             patterns = [patterns]
         }
 
+        // Check we have an array
```

Comment:
No need to state the obvious ;)

Reply:
Agree here, either add more context or remove comment completely